### ISSUE TITLE
ncurses messing up tmux screen after context switches

### ISSUE THREAD

**Issue by marknelson** [Microsoft/console#242]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)

Microsoft Windows [Version 10.0.17738.1000]

* What you're doing and what's happening: (Copy & paste specific commands and their output, or include screen shots)

The bottom lines of my screen get foozled, typically after a context switch in which they are redrawn. Here's a typical example in which the tmux status line somehow got duplicated, and an additional blank line was added. This is in full screen mode, no resizing has occurred, and nothing was being written to the screen - the problem just popped up after switching from another app back to the console:

![messed up](https://i.imgur.com/bdYdf4B.png)

This is in a multi-monitor setup, but I've seen exactly the same problem on a single-monitor setup.

When I look at the screen, sometimes I see a cursor down at the bottom - seems like it is in a funny position and just ready to cause trouble:

![a](https://i.imgur.com/Rrwcl7u.png)

* What's wrong / what should be happening instead:

Bottom line should not be corrupted!

I've seen a few other console corruption issues listed here, but none of them seem to match up exactly with this.

Note that I don't have exact steps to reproduce this, but it happens constantly. If needed we can probably create a recipe to reproduce it.

-Mark

**Comment by zadjii-msft** [comment]
Yea, I've seen this before. It's not particularly hard to reproduce, but it's still pretty hard to debug who's exactly at fault. There's some real janky code that prevents the cursor from moving to the next line when it's right at the end of a line until the next character is actually output. I'd think that code is probably not playing well with our resize handling, and we're placing the cursor on the next line (even though it shouldn't be) during a resize, so when we come back from the resize, the cursor has forgotten it shouldn't be there.

Actually, that might be it. Thanks for the help [rubber duck debugging!
](https://en.wikipedia.org/wiki/Rubber_duck_debugging)
This is already being tracked internally as msft:16861099. Now I have a hunch of where to look next :)

**Comment by marknelson** [comment]
>during a resize

I want to emphasize that I'm seeing this in a full screen console, and no resizing is taking place. So fixing problems in resize may or may not help here. Maybe that's what you're saying, just want to be sure I have it down.

**Comment by miniksa** [comment]
This is in Insider builds 18312+.
